### DISCUSSION TITLE
Docker: (RuntimeError) environment variable DATABASE_HOST is missing.

### DISCUSSION THREAD

**Discussion by LinasKo** [zoonk/uneebee#172] [Bugs & Issues] [answered]
```
* Repo: Uneebee
* Host System: Ubuntu 22.04.3 LTS
* Docker: Docker version 24.0.7, build afdd53b
```

I build the docker image, and tried tu run it, but it seems the database isn't fully set up, throwing the error below.

Commands I ran:
```bash
docker build -t uneebee 
docker run uneebee
```

Error raised:
```
=ERROR REPORT==== 23-Nov-2023::12:36:50.024015 ===
Invalid -name given to erl, 
=ERROR REPORT==== 23-Nov-2023::12:36:50.024015 ===
Invalid -name given to erl, 
ERROR! Config provider Config.Reader failed with:
** (RuntimeError) environment variable DATABASE_HOST is missing.

    /app/releases/0.1.0/runtime.exs:26: (file)
    (elixir 1.15.6) src/elixir.erl:376: anonymous fn/4 in :elixir.eval_external_handler/1
    (stdlib 5.0.2) erl_eval.erl:750: :erl_eval.do_apply/7
    (stdlib 5.0.2) erl_eval.erl:494: :erl_eval.expr/6
    (stdlib 5.0.2) erl_eval.erl:136: :erl_eval.exprs/6
    (elixir 1.15.6) src/elixir.erl:361: :elixir.eval_forms/4
    (elixir 1.15.6) lib/module/parallel_checker.ex:112: Module.ParallelChecker.verify/1
    (elixir 1.15.6) lib/code.ex:543: Code.validated_eval_string/3

Runtime terminating during boot ({,[{elixir_eval,__FILE__,1,[{_},{_}]},{elixir,-eval_external_handler/1-fun-2-,4,[{_},{_},{_}]},{erl_eval,do_apply,7,[{_},{_}]},{erl_eval,expr,6,[{_},{_}]},{erl_eval,exprs,6,[{_},{_}]},{elixir,eval_forms,4,[{_},{_}]},{Elixir.Module.ParallelChecker,verify,1,[{_},{_}]},{Elixir.Code,validated_eval_string,3,[{_},{_}]}]})

Crash dump is being written to: erl_crash.dump...done
```

Are there further setup requirements before using the docker image? E.g. I cannot see the `mix setup` inside the Dockerfile that's mentioned in the install instructions, even though at least some of its component methods from `mix.exs` are being called.

I've never tried Elixir or Phoenix so would be keen to get Docker set up instead of tackling the learning curve of those. Sounds like a very interesting project overall - got a friend on a lookout to make a high-end English course, and was hoping to demo the project to them :)

**Comment by ceolinwill** [comment]
Sorry, I'm using that `Dockerfile` to deploy the demo app to fly.io. You'd need to have a Postgres instance running and pass some environment variables. It's not meant to be used for a local dev server.

Unfortunately, I haven't created a `Dockerfile` for running UneeBee locally yet. I've [updated the README](https://github.com/zoonk/uneebee?tab=readme-ov-file#getting-started) to clarify that. I've also created a new issue to implement this before v1: https://github.com/zoonk/uneebee/issues/60 (PRs are welcome!).

In the meantime, the workaround is installing Elixir on your machine and following the [getting started instructions](https://github.com/zoonk/uneebee?tab=readme-ov-file#getting-started).

**Reply by LinasKo** [reply]
That makes sense, thanks!

I've used fly.io before - would you say it's a straightforward process to deploy to Uneebee there, or is the local deployment the easiest option at the moment?

**Reply by ceolinwill** [reply]
It's kinda straightforward but you might run into some issues and have to do some debugging and I didn't have the time to document all the steps required to make it work properly in production. I'll do before v1. For example, you'll need to configure Cloudflare R2 for storage, Resend for sending emails, a Postgres database (I'm using [Neon](https://neon.tech/)).

If it's just for a quick demo and you're not feeling like debugging things, running it locally might be the easiest option (especially if you're running it on MacOS or Linux. I haven't tested it running on Windows, so I'm not sure if it works properly).

If you do try to run it, please share the roadblocks, either here or opening a PR to update the docs.

**Reply by ceolinwill** [reply]
I forgot to mention. If you try do deploy to fly.io, you might want to have a look at [this file](https://github.com/zoonk/uneebee/blob/main/fly.toml) and change the values under `env` and `build.args`. Or just delete that file and run `fly deploy`. Fly will create that file for you and even set up a Postgres database, I think.

**Comment by ceolinwill** [comment, marked as answer]
@LinasKo A PR for this has been merged: https://github.com/zoonk/uneebee/pull/128
